### PR TITLE
Demo scripts for m1 facilitator

### LINKED ISSUE #365: Demo scripts for m1 facilitator

### PR DESCRIPTION
**Pull request by jayeshbairagi**
Fixes #365

### CODE REVIEW

File: demo/m1_facilitator/bin/request_deposit_testnet.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      type: 'string',
+      name: 'web3EndPoint',
+      default: 'https://rpc.slock.it/goerli',
+      message: 'Enter origin chain(goerli) end point:',
+    },
+    {
+      type: 'string',
+      name: 'valueTokenAddress',
+      default: '0xd426b22f3960d01189a3d548b45a7202489ff4de',
+      message: 'Enter value token address:',
```
Comment (deepesh-kn) [inline demo/m1_facilitator/bin/request_deposit_testnet.ts]: Why are we using `value token`  term?
Why not `ERC20 token`?

File: demo/m1_facilitator/bin/request_deposit_testnet.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      message: 'Enter amount to deposit in (wei):',
+      validate(input: string): number {
+        return new BN(input).gtn(0);
+      },
+    },
+    {
+      type: 'string',
+      name: 'beneficiary',
+      message: 'Enter beneficiary address on the metachain(Hadapsar-1405):',
+      validate(input: string): string {
```
Comment (deepesh-kn) [inline demo/m1_facilitator/bin/request_deposit_testnet.ts]: Add `depositor address` as a beneficiary address as a default.

File: demo/m1_facilitator/bin/request_withdraw_testnet.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    },
+    {
+      type: 'password',
+      name: 'password',
+      message: 'Enter withdrawer keystore password:',
+    },
+    {
+      type: 'string',
+      name: 'amountToWithdraw',
+      message: 'Enter amount to deposit in (wei):',
```
Comment (deepesh-kn) [inline demo/m1_facilitator/bin/request_withdraw_testnet.ts]: withdraw amount in atto

File: demo/m1_facilitator/bin/request_deposit_testnet.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    },
+    {
+      type: 'password',
+      name: 'password',
+      message: 'Enter depositor keystore password:',
+    },
+    {
+      type: 'string',
+      name: 'amountToDeposit',
+      message: 'Enter amount to deposit in (wei):',
```
Comment (deepesh-kn) [inline demo/m1_facilitator/bin/request_deposit_testnet.ts]: in atto

File: demo/m1_facilitator/bin/create_depositor_withdrawer.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      when: (response: any): boolean => response.actor === 'withdrawer',
+    },
+  ]);
+
+  const web3 = new Web3();
+  const ethereumAccount = web3.eth.accounts.create(web3.utils.randomHex(8));
+
+  if (answer.shouldFundDepositor === 'y') {
+    await fundFromFaucet(ethereumAccount.address, '5');
+    console.log(`✅ Funded ${100} OST to address ${ethereumAccount.address}`);
```
Comment (deepesh-kn) [inline demo/m1_facilitator/bin/create_depositor_withdrawer.ts]: Do not use hardcode value here.

**Review by deepesh-kn — CHANGES_REQUESTED** [review]
1. Please remove the images, add it as a preformatted text from the terminal.
2. Add try/catch in the code, if it breaks right now it gives no output
3. Please see inline comments.

File: demo/m1_facilitator/bin/request_deposit_testnet.ts (hunk 1 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/request_deposit_testnet.ts]: I wanted to be specific to the terms we have been using throughout mosaic-1 and that is why I used `value token`.
Thank you for pointing it out now, for the user-doc I can update it to `ERC20 token`.

File: demo/m1_facilitator/bin/create_depositor_withdrawer.ts (hunk 5 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/create_depositor_withdrawer.ts]: Should we then fetch the values using contract instance and web3? Or just leave out the funded amount value?

File: demo/m1_facilitator/bin/request_deposit_testnet.ts (hunk 2 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/request_deposit_testnet.ts]: The path to the Keystore is taken using the inquirer. So, till this inquirer prompt is not completed we won't get the address(using the Keystore). 
I think we can keep it this way only.

File: demo/m1_facilitator/bin/request_deposit_testnet.ts (hunk 1 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/request_deposit_testnet.ts]: Updated

File: demo/m1_facilitator/bin/request_deposit_testnet.ts (hunk 2 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/request_deposit_testnet.ts]: Handled it using another inquirer to get the default value of the beneficiary address.

File: demo/m1_facilitator/bin/request_withdraw_testnet.ts (hunk 3 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/request_withdraw_testnet.ts]: Updated

File: demo/m1_facilitator/bin/request_deposit_testnet.ts (hunk 4 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/request_deposit_testnet.ts]: Updated

File: demo/m1_facilitator/bin/create_depositor_withdrawer.ts (hunk 5 above)
Comment (jayeshbairagi) [reply demo/m1_facilitator/bin/create_depositor_withdrawer.ts]: Updated

**Review by deepesh-kn — APPROVED** [review]

**Pull request merged**